src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -149,21 +157,32 @@ public void start(Map<String, String> properties) {
                 timestampColumns
             );
           }
-          partition = Collections.singletonMap(
-              JdbcSourceConnectorConstants.TABLE_NAME_KEY,
-              tableOrQuery
-          );
+          tablePartitionsToCheck = tableToPartitions.get(tableOrQuery);
           break;
         case QUERY:
           partition = Collections.singletonMap(
               JdbcSourceConnectorConstants.QUERY_NAME_KEY,
               JdbcSourceConnectorConstants.QUERY_NAME_VALUE
           );
+          tablePartitionsToCheck = new ArrayList<>();
+          tablePartitionsToCheck.add(partition);
```

Comment:
Suggestion to replace these two lines with:

    tablePartitionsToCheck = Collections.singleton(partition);